server/service/client_hosts.go

```diff
@@ -99,10 +103,14 @@ func (c *Client) TransferHosts(hosts []string, label string, status, searchQuery
 		return err
 	}
 
+	var teamIDPtr *uint
+	if teamID != 0 {
+		teamIDPtr = &teamID
+	}
 	if len(hosts) != 0 {
 		verb, path := "POST", "/api/latest/fleet/hosts/transfer"
 		var responseBody addHostsToTeamResponse
-		params := addHostsToTeamRequest{TeamID: ptr.Uint(teamID), HostIDs: hostIDs}
+		params := addHostsToTeamRequest{TeamID: teamIDPtr, HostIDs: hostIDs}
```

Comment:
What is `teamIDPtr` if teamID above is 0?

Reply:
`teamIDPtr` is initialized to `nil`